spyder/app/mainwindow.py

```diff
@@ -2969,10 +2972,23 @@ def open_external_file(self, fname):
         variable explorer inside Spyder.
         """
         fname = encoding.to_unicode_from_fs(fname)
-        if osp.isfile(fname):
-            self.open_file(fname, external=True)
-        elif osp.isfile(osp.join(CWD, fname)):
-            self.open_file(osp.join(CWD, fname), external=True)
+        if osp.exists(osp.join(CWD, fname)):
+            fpath = osp.join(CWD, fname)
+        elif osp.exists(fname):
+            fpath = fname
+        else:
+            return
+
+        if osp.isfile(fpath):
+            self.open_file(fpath, external=True)
+        elif osp.isdir(fpath):
+            QMessageBox.warning(
+                None, "Spyder",
+                _('Not opening folder <code>{fpath}</code> . To open that '

```

Comment:
Maybe we should phrase this message like:
```
Spyder didn't open the folder <code>{fpath}</code>. To open a folder 
as a project, please use <code>spyder -p "{fpath}"</code>.
```
what do you think @ccordoba12 ?

Reply:
Nice improvement @dalthviz! My suggestion:

```
To open <code>{fpath}</code> as a project with Spyder, 
please use <code>spyder -p "{fpath}"</code>.
```